### PR TITLE
Fix #1702: Add click functionality to OngoingTopicList item

### LINKED ISSUE #1702: Make ongoing topic in OngoingTopicList clickable
Currently `topics` in `OngoingTopicList` are not clickable.

They should be made clickable and should route to `Topic Info Page`

Reference: https://github.com/oppia/oppia-android/issues/1659#issuecomment-675080309

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fix #1702 
As per this comment #1659 (comment)

The ongoing-topic-item item should be clickable and should route to Topic-Info page.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@miaboloix @BenHenning can you have a look at this error https://github.com/oppia/oppia-android/pull/1705/checks?check_run_id=1021744130

Not sure what should be the exact reason/solution for this error.

**Comment by anandwana001** [comment]
@miaboloix Whenever we create a new ViewModel or update it from the current codebase do we require to build it?

`~/bazel-databinding-new/bazel-bin/src/bazel build //:view_models --android_databinding_use_v3_4_args --experimental_android_databinding_v2 --override_repository=android_tools=/tmp/android_tools --java_header_compilation=false`

File: app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+class OngoingTopicItemViewModel(
+  val activity: AppCompatActivity,
+  val internalProfileId: Int,
+  val topic: Topic,
+  val entityType: String,
+  private val intentFactoryShim: IntentFactoryShim
+) :
+  ObservableViewModel(),
+  RouteToTopicListener {
+  fun onTopicItemClicked() {
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt]:
```suggestion
) : ObservableViewModel(), RouteToTopicListener {

  fun onTopicItemClicked() {
```

File: app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
 import androidx.lifecycle.ViewModel
+import org.oppia.app.home.RouteToTopicListener
 import org.oppia.app.model.Topic
+import org.oppia.app.shim.IntentFactoryShim
+import org.oppia.app.viewmodel.ObservableViewModel
 
 /** [ViewModel] for displaying topic item in [OngoingTopicListActivity]. */
-class OngoingTopicItemViewModel(val topic: Topic, val entityType: String) : ViewModel()
+class OngoingTopicItemViewModel(
+  val activity: AppCompatActivity,
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt]: Here & elsewhere: please make values private if they don't need to be accessed outside their contained classes.

**Review by BenHenning — APPROVED** [review]
Thanks @rt4914. The solution LGTM (feel free to resolve the open comments yourself before merging), though #1702 appears to already be fixed?

**Comment by BenHenning** [comment]
@rt4914 you will probably need to add RouteToTopicListener to https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel#L43. That should fix the build issue.

**Comment by rt4914** [comment]
> @rt4914 you will probably need to add RouteToTopicListener to https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel#L43. That should fix the build issue.

Okay, thanks. I will try this.

File: app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt]: Updated.

File: app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/ongoingtopiclist/OngoingTopicItemViewModel.kt]: Updated.

**Pull request merged**